pkg/apis/tenancy/v1alpha1/types.go

```diff
@@ -629,3 +630,147 @@ var (
 		},
 	}
 )
+
+const (
+	// ThisWorkspaceName is the name of the ThisWorkspace singleton.
+	ThisWorkspaceName = "this"
+
+	// ThisWorkspaceFinalizer attached to new ClusterWorkspace (in phase ClusterWorkspacePhaseScheduling) resources so that we can control
+	// deletion of ThisWorkspace resources
+	ThisWorkspaceFinalizer = "tenancy.kcp.dev/thisworkspace"
+)
+
+// ThisWorkspace describes the current workspace.
+//
+// +crd
```

Comment:
nit: we're doing this cleanup slowly, but can we put all of the generator tags in a separate comment above the GoDoc, with an empty line between them? This will still get the tags to apply but will not put them into GoDoc

Reply:
(this applies for the types and the fields as well)

Reply:
and split up the files per kind

Reply:
did the split https://github.com/p0lyn0mial/kcp/pull/8

The comment changes we should do everywhere outside of this PR.